lib/HTTP/Cookies.pm

```diff
@@ -431,9 +431,10 @@ sub save
 {
     my $self = shift;
     my $file = shift || $self->{'file'} || return;
+    my $ignore_discard = ( @_ ? shift : $self->{'ignore_discard'} );
     open(my $fh, '>', $file) or die "Can't open $file: $!";
     print {$fh} "#LWP-Cookies-1.0\n";
-    print {$fh} $self->as_string(!$self->{ignore_discard});
+    print {$fh} $self->as_string(!ignore_discard);
```

Comment:
We have a bareword here rather than a variable.

`print {$fh} $self->as_string(!$ignore_discard);`

Reply:
I've rebased and rewritten the commit with this correction.